.github/workflows/test.yml

```diff
@@ -18,22 +16,17 @@ jobs:
       - name: Install Nix
         uses: cachix/install-nix-action@v21
         with:
-          nix_path: nixpkgs=channel:nixos-unstable
-          install_url: https://releases.nixos.org/nix/nix-2.10.3/install
+          nix_path: nixpkgs=channel:nixos-2305
+          install_url: https://releases.nixos.org/nix/nix-2.16.0/install
           extra_nix_config: |
-            allowed-uris = ${{ env.ALLOWED_URIS }}
             trusted-public-keys = ${{ env.TRUSTED_PUBLIC_KEYS }}
             substituters = ${{ env.SUBSTITUTERS }}
             experimental-features = nix-command flakes
-      - name: Setup Cachix
```

Comment:
I really think we can do without cachix. 
Everything is cached on hydra anyway.
Having cachix might actually be slowing things down due to upload overheads.
I'm testing this workflow on my fork and it runs quick/same-time https://github.com/zeme-wana/marlowe-cardano/actions